src/wallet/wallet2.cpp

```diff
@@ -3464,14 +3570,22 @@ bool wallet2::check_connection(uint32_t *version, uint32_t timeout)
 //----------------------------------------------------------------------------------------------------
 bool wallet2::generate_chacha_key_from_secret_keys(crypto::chacha_key &key) const
 {
-  const account_keys &keys = m_account.get_keys();
-  const crypto::secret_key &view_key = keys.m_view_secret_key;
-  const crypto::secret_key &spend_key = keys.m_spend_secret_key;
-  tools::scrubbed_arr<char, sizeof(view_key) + sizeof(spend_key) + 1> data;
-  memcpy(data.data(), &view_key, sizeof(view_key));
-  memcpy(data.data() + sizeof(view_key), &spend_key, sizeof(spend_key));
-  data[sizeof(data) - 1] = CHACHA8_KEY_TAIL;
-  crypto::generate_chacha_key(data.data(), sizeof(data), key);
+  ledger::Device &device =  m_account.get_device();
+  if (device) {
+    char prekey[200];
+    device.get_chacha8_prekey(prekey);
+    crypto::generate_chacha_key(&prekey[0], sizeof(prekey), key, true);
+    memset(prekey, 0, sizeof(prekey));
```

Comment:
The compiler *will* optimize out this dead memset, leaving that key data on the stack.

Use `scrubbed_arr<char, 200> prekey` instead.

Reply:
OK, noted.